src/core/package_cache.cpp

```diff
@@ -117,8 +117,9 @@ namespace mamba
         {
             fs::path tarball_path = m_path / s.fn;
             // validate that this tarball has the right size and MD5 sum
-            valid = validate::file_size(tarball_path, s.size);
-            valid = (valid || s.size == 0) && validate::md5(tarball_path, s.md5);
+            // we handle the case where s.size == 0 (explicit packages) or md5 is unknown
+            valid = s.size == 0 || validate::file_size(tarball_path, s.size);
+            valid = valid && (s.md5.empty() || validate::md5(tarball_path, s.md5));
```

Comment:
I'm not sure we should authorize loose validation. Here missing file size and MD5 would return valid, which is not the case (it's `unknown`).

We could/should then have a `tarball_safety_checks`, with possible values `all`, `one_of`, `size`, `md5`, `none`/`disabled`, and explicitly set it to a given value in corner cases (explicit spec installation, etc.) with probably a warning log message

for now, maybe something like the following would be preferable?
```cpp
bool unknown_size = s.size == 0;
bool valid_size = validate::file_size(tarball_path, s.size);

bool unknown_md5 = s.md5.empty();
bool valid_md5 = validate::md5(tarball_path, s.md5);

valid = (valid_size || unknown_size) && (valid_size || unknown_size)  && (valid_size || valid_md5);
```